packages/shared-core/src/filters.ts

```diff
@@ -329,20 +397,28 @@ export const runLuceneQuery = (docs, query) => {
  * @param sortOrder the sort order ("ascending" or "descending")
  * @param sortType the type of sort ("string" or "number")
  */
-export const luceneSort = (docs, sort, sortOrder, sortType = "string") => {
+export const luceneSort = (
+  docs: any[],
+  sort: string | number,
+  sortOrder: SortDirection,
+  sortType = "string"
+) => {
   if (!sort || !sortOrder || !sortType) {
     return docs
   }
-  const parse = sortType === "string" ? x => `${x}` : x => parseFloat(x)
-  return docs.slice().sort((a, b) => {
-    const colA = parse(a[sort])
-    const colB = parse(b[sort])
-    if (sortOrder === "Descending") {
-      return colA > colB ? -1 : 1
-    } else {
-      return colA > colB ? 1 : -1
-    }
-  })
+  const parse =
+    sortType === "string" ? (x: any) => `${x}` : (x: string) => parseFloat(x)
+  return docs
+    .slice()
+    .sort((a: { [x: string]: any }, b: { [x: string]: any }) => {
+      const colA = parse(a[sort])
+      const colB = parse(b[sort])
+      if (sortOrder.toLowerCase() === "descending") {
```

Comment:
Case change in `"descending"`. Should a constant / enum be used here for safety?

Reply:
We have a mix of usages between js and ts, so we can not be type-safe. For this I check case insensitive